### DISCUSSION TITLE
How to use multiple custom fonts

### DISCUSSION THREAD

**Discussion by baswarmerdam** [dlemstra/Magick.NET#1752] [Help] [answered]
Hi all,

I have multiple .otf files that I want to use to add text to an image. I load the custom font by embedding it as a resource, writing it to a temp path. This does NOT work if I use it like this:

`drawables.Font(myCustomFont).Text("Hello world!")`

It will just use the default font.
However, if I use `MagickReadSettings` it DOES work:

```
using var image = new MagickImage(MagickColors.White, 800, 600));
image.Settings.Font = myCustomFont;
image.Settings.FontFamily = myCustomFont;
```

Now everything I draw is using the custom font that I loaded (which also means that the issue is not with loading the font).
I am not sure this works because I set the FontFamily instead of the font, but if it is, I could not find how to do this using `Drawables`.

So my issue is: since using `.Font(myOtherFont)` to switch to another font does not work for me, how can I switch to other fonts? I guess in theory I could maybe create multiple images and use `Composite` to merge them or something? Feels unnecessary complex though, I feel like I am missing something obvious here.

.NET 8
Windows
14.0

**Comment by dlemstra** [comment]
Can you create a small project on GitHub that demonstrates this issue so I can give you a better answer?

**Reply by baswarmerdam** [reply]
@dlemstra Certainly: https://github.com/baswarmerdam/MagickNetFontTest

And thank you.

**Reply by dlemstra** [reply, marked as answer]
The problem in your case is that your temporary file doesn't contain a `.ttf` or `.otf` extension so ImageMagick thinks it's a really weird font name and tries to load that font. There is probably a slightly different check in the read settings that allows these kind of font names. To solve this for your situation you can do this instead:

```C#
string tempFontPath = $@"{Path.GetTempPath()}\{Guid.NewGuid}.{resourceName}";
```

**Reply by baswarmerdam** [reply]
Awesome. Thank you so much!
